build.gradle

```diff
@@ -2441,13 +2459,18 @@ project(':tools') {
     implementation libs.jacksonDataformatCsv
     implementation libs.jacksonJDK8Datatypes
     implementation libs.slf4jApi
-    implementation libs.slf4jReload4j
+    implementation libs.slf4jLog4j2
+    implementation libs.log4j2Api
+    implementation libs.log4j2Core
+    implementation libs.log4j1Bridge2Api
     implementation libs.joptSimple
     implementation libs.re2j
 
     implementation libs.jose4j                    // for SASL/OAUTHBEARER JWT validation
     implementation libs.jacksonJakartarsJsonProvider
 
+    compileOnly libs.spotbugs
```

Comment:
Why is this needed?

Reply:
I added `spotbugs` to address the compiler warnings. While we could suppress these warnings using `-Xlint:all,-classFile`, I prefer not to relax compiler checks.

Ideally, we should avoid depending on core-specific methods altogether. However, that would require thorough pass to our code. I suggest we create a separate JIRA ticket to track this architectural improvement as a future enhancement.

For further details: https://github.com/apache/logging-log4j2/issues/2144#issue-2061148244

Reply:
At Apache Logging we had several other issue reports regarding our usage of annotations in the `provided` scope (see https://github.com/apache/logging-log4j2/issues/3110#issuecomment-2423586754 for example).

Regarding the [`@SuppressFBWarnings` annotation](https://javadoc.io/doc/com.github.spotbugs/spotbugs-annotations/latest/edu/umd/cs/findbugs/annotations/SuppressFBWarnings.html) that causes this particular problem:

- IMHO the compiler should not issue any warnings if it is missing, since the annotation has a retention of `CLASS` and is totally invisible at runtime. I submitted [JDK-8342833](https://bugs.openjdk.org/browse/JDK-8342833) to change the compiler's behavior.
- Log4j Core could theoretically move Spotbugs Annotations from the `provided` to the `compile` scope, but this could cause legal problems, since the annotation library is licensed under LGPL and this can not be changed (see https://github.com/spotbugs/spotbugs/issues/3144). This is one of the reasons we keep the library in the `provided` scope, so it does not propagate to consumers.

Reply:
Yeah, the licensing issue is the reason I wanted to avoid using this library at all. Can we use a different library for annotations, one that is Apache licensed?